### PR TITLE
[Chore] Modernize project

### PR DESCRIPTION
**Pull request by kittinunf**
- up versions
  - upgrade Fuel version (able to use stetho now) 🎉🎉

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -38,35 +38,35 @@ android {
...
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$Versions.Tools.kotlin"
-    implementation 'androidx.appcompat:appcompat:1.1.0-alpha01'
-    implementation 'androidx.recyclerview:recyclerview:1.1.0-alpha01'
-    implementation 'androidx.constraintlayout:constraintlayout:2.0.0-alpha3'
+
+    //android
+    implementation "androidx.appcompat:appcompat:$Versions.androidX"
+    implementation "androidx.recyclerview:recyclerview:$Versions.androidX"
+    implementation "androidx.constraintlayout:constraintlayout:$Versions.constraintLayout"
     implementation 'androidx.paging:paging-runtime:2.1.0-rc01'
```
Comment (worker8-1) [inline app/build.gradle]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/worker8/Pixels/blob/cd00bd156b5acad31bc778d24ad8b96a41fefaca/app/build.gradle#L48"></span>A newer version of androidx.paging:paging-runtime than 2.1.0-rc01 is available: 2.1.0</td>
    </tr>
  </tbody>
</table>

File: app/build.gradle [hunk 2]
```diff
@@ -38,35 +38,35 @@ android {
...
+    implementation 'androidx.lifecycle:lifecycle-extensions:2.1.0-alpha01'
```
Comment (worker8-1) [inline app/build.gradle]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/worker8/Pixels/blob/cd00bd156b5acad31bc778d24ad8b96a41fefaca/app/build.gradle#L49"></span>A newer version of androidx.lifecycle:lifecycle-extensions than 2.1.0-alpha01 is available: 2.1.0-alpha02</td>
    </tr>
  </tbody>
</table>

**Comment by worker8-1** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          9 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/cd00bd156b5acad31bc778d24ad8b96a41fefaca/app/src/main/res/drawable/ic_logo_foreground.xml#L15">app/src/main/res/drawable/ic_logo_foreground.xml#L15</a> - Attribute <code>endX</code> is only used in API level 24 and higher (current min is 21)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/cd00bd156b5acad31bc778d24ad8b96a41fefaca/app/src/main/res/drawable/ic_logo_foreground.xml#L16">app/src/main/res/drawable/ic_logo_foreground.xml#L16</a> - Attribute <code>endY</code> is only used in API level 24 and higher (current min is 21)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/cd00bd156b5acad31bc778d24ad8b96a41fefaca/app/src/main/res/drawable/ic_logo_foreground.xml#L17">app/src/main/res/drawable/ic_logo_foreground.xml#L17</a> - Attribute <code>startX</code> is only used in API level 24 and higher (current min is 21)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/cd00bd156b5acad31bc778d24ad8b96a41fefaca/app/src/main/res/drawable/ic_logo_foreground.xml#L18">app/src/main/res/drawable/ic_logo_foreground.xml#L18</a> - Attribute <code>startY</code> is only used in API level 24 and higher (current min is 21)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/cd00bd156b5acad31bc778d24ad8b96a41fefaca/app/src/main/res/drawable/ic_logo_foreground.xml#L22">app/src/main/res/drawable/ic_logo_foreground.xml#L22</a> - Attribute <code>offset</code> is only used in API level 24 and higher (current min is 21)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/cd00bd156b5acad31bc778d24ad8b96a41fefaca/app/src/main/res/drawable/ic_logo_foreground.xml#L25">app/src/main/res/drawable/ic_logo_foreground.xml#L25</a> - Attribute <code>offset</code> is only used in API level 24 and higher (current min is 21)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/cd00bd156b5acad31bc778d24ad8b96a41fefaca/app/src/main/res/drawable/ic_outline_bug.xml#L8">app/src/main/res/drawable/ic_outline_bug.xml#L8</a> - Very long vector path (852 characters), which is bad for performance. Considering reducing precision, removing minor details or rasterizing vector.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/cd00bd156b5acad31bc778d24ad8b96a41fefaca/app/src/main/res/layout/comment_header.xml#L7">app/src/main/res/layout/comment_header.xml#L7</a> - Possible overdraw: Root element paints background <code>?pixelBackgroundColor</code> with a theme that also paints a background (inferred theme is <code>@style/AppTheme</code>)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/cd00bd156b5acad31bc778d24ad8b96a41fefaca/app/src/main/res/mipmap-hdpi/ic_launcher_round.png#L0">app/src/main/res/mipmap-hdpi/ic_launcher_round.png#L0</a> - Launcher icon used as round icon did not have a circular shape</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/build.gradle [hunk 3]
```diff
@@ -38,35 +38,35 @@ android {
...
+    implementation "androidx.recyclerview:recyclerview:$Versions.androidX"
     implementation 'com.google.android.material:material:1.1.0-alpha03'
-//    implementation "com.squareup.retrofit2:retrofit:$Versions.retrofit"
-//    implementation "com.squareup.retrofit2:adapter-rxjava2:$Versions.retrofit"
-//    implementation "com.squareup.retrofit2:converter-gson:$Versions.retrofit"
+
+    //deps
     implementation "com.github.bumptech.glide:glide:$Versions.glide"
     kapt "com.github.bumptech.glide:compiler:$Versions.glide"
-    implementation "com.facebook.stetho:stetho:$Versions.stetho"
```
Comment (worker8) [inline app/build.gradle]: I want to use stetho, can we keep this as `debugImplementation`?

File: app/src/main/java/beepbeep/pixelsforredditx/PixelApplication.kt [hunk 4]
```diff
@@ -2,14 +2,12 @@ package beepbeep.pixelsforredditx
...
 import android.app.Application
 import android.content.Context
-import com.facebook.stetho.Stetho
 
 class PixelApplication : Application() {
 
     override fun onCreate() {
         super.onCreate()
         _applicationContext = this
-        Stetho.initializeWithDefaults(this)
```
Comment (worker8) [inline app/src/main/java/beepbeep/pixelsforredditx/PixelApplication.kt]: so... also keep this 💡

File: app/build.gradle [hunk 5]
```diff
@@ -38,35 +38,35 @@ android {
...
-
-    implementation "com.squareup.picasso:picasso:$Versions.picasso"
-    implementation 'androidx.lifecycle:lifecycle-extensions:2.1.0-alpha01'
+    implementation "com.github.kittinunf.fuel:fuel-rxjava:$Versions.fuel"
+    implementation "com.github.kittinunf.result:result:$Versions.result"
     implementation "com.github.worker8:HeaderFooterAdapter:$Versions.headerFooterAdapter"
+    implementation "com.jakewharton.rxbinding3:rxbinding:$Versions.rxBinding"
+    implementation "com.squareup.picasso:picasso:$Versions.picasso"
```
Comment (worker8) [inline app/build.gradle]: can you also remove Picasso?
_I added it and soon decided to go with Glide_

File: app/build.gradle (hunk 3 above)
Comment (kittinunf) [reply app/build.gradle]: we are gonna use the one from Fuel?

**Comment by worker8** [comment]
Looks good, go go go 🚀

**Pull request merged**